frontend/web/components/App.js

```diff
@@ -350,12 +351,27 @@ const App = class extends Component {
       return <div>{this.props.children}</div>
     }
     const announcementValue = Utils.getFlagsmithJSONValue('announcement', null)
+    const announcementPerPageDismissed = flagsmith.getTrait(
+      'dismissed_announcement_per_page',
+    )
+    const announcementPerPageValue = Utils.getFlagsmithJSONValue(
+      'announcement_per_page',
+      null,
+    )
+    const showAnnouncementPerPage =
+      !announcementPerPageDismissed ||
+      announcementPerPageDismissed !== announcementPerPageValue.id
+    Utils.getFlagsmithHasFeature('announcement_per_page') &&
+      announcementPerPageValue?.pages?.length > 0
     const dismissed = flagsmith.getTrait('dismissed_announcement')
     const showBanner =
       announcementValue &&
       (!dismissed || dismissed !== announcementValue.id) &&
       Utils.getFlagsmithHasFeature('announcement') &&
       this.state.showAnnouncement
+    const announcementInPage = announcementPerPageValue?.pages?.some((page) =>
```

Comment:
Hmm, would this work for urls with ids or does this need to match in a more detailed way?

Reply:
I understand this should target a specific page and not particular entities.
Maybe in the future if we need it at some point.

Reply:
Hmm I don't understand how this will work, how would this target a page in remote config?

Nearly every page has ids in the pathname e.g. 
'/project/x/environment/y/features'